File: fib-gen.lua
local n1 = 1
local n2 = 2
local iter = 0
print('Fibonacci Sequence Length: ')
local numberOfIterations = io.read("*n")

while iter <= numberOfIterations do
  print("Fib(".. iter .. ")=" .. n1)
  -- the-strf-way: print(string.format("Fib(%s)=%d",iter,n1))
  new_number = n1 + n2
  n1 = n2
  n2 = new_number
  iter = iter + 1
end

File: fib-gen.go
package main

import "fmt"

func main() {
  fmt.Println("Fibonacci Sequence Length: ")
  var numberOfIterations int
  fmt.Scanf("%v", &numberOfIterations)
  var n1, n2 int = 1, 2
  iter := 0
  for iter <= numberOfIterations {
    fmt.Printf("Fib(%v)=%v\n",iter,n1)
    new_number := n1 + n2
    n1 = n2
    n2 = new_number
    iter++
  }
}